### PR TITLE
update concepts guides

### LINKED ISSUE #4308: orbs, jobs, steps, workflows
# Locations
https://circleci.com/docs/2.0/jobs-steps/#section=getting-started

(It's possible we should split this page up now I'm thinking about it, maybe Workflows separately)

# Conflict Description
Screenshot 2020-04-20 at 16 42 26

This ^^ section describes the container plan usage, which is a legacy plan I believe.

Also when talking about Jobs and Steps, use of  "greater" and "more than" is confusing to me, and I wonder if it's meaning greater than in 1.0? If so we should change this language I think.

Screenshot 2020-04-20 at 16 46 54

This ^^ section needs updating too I think. Not sure about the reuse of machine reruns part, and it could read easier overall.

### PR DESCRIPTION
**Pull request by rosieyohannan**
fixes #4308 

* tabbed 2.1/2 versions of code snippets in concepts guide 
* remove some out of date elements.
* other editorial fixes

### CODE REVIEW

File: jekyll/_cci2/concepts.md [hunk 1]
```diff
@@ -89,54 +89,89 @@ Jobs are the building blocks of your config. Jobs are collections of [steps](#st
...
+ build3:
+   macos: # Specifies a macOS virtual machine with Xcode version 11.3
+     xcode: "11.3.0"
 # ...          
- ```
+```
+
+{:.tab.executors.Server}
+```yaml
+version: 2.0
```
Comment (thomasr8) [inline jekyll/_cci2/concepts.md:131]:
```suggestion
version: 2.1
```

File: jekyll/_cci2/concepts.md [hunk 2]
```diff
@@ -89,54 +89,89 @@ Jobs are the building blocks of your config. Jobs are collections of [steps](#st
...
 
 When using the docker executor and running docker commands, the `setup_remote_docker` key can be used to spin up another docker container in which to run these commands, for added security. For more information see the [Running Docker Commands]({{ site.baseurl }}/2.0/building-docker-images/#accessing-the-remote-docker-environment) guide.
 
 #### Steps
-{:.no_toc}
 
 Steps are actions that need to be taken to complete your job. Steps are usually a collection of executable commands. For example, the [`checkout`]({{ site.baseurl }}//2.0/configuration-reference/#checkout) step, which is a _built-in_ step available across all CircleCI projects, checks out the source code for a job over SSH. Then, the `run` step allows you to run custom commands, such as executing the command `make test`  using a non-login shell by default. Commands can also be defined [outside the job declaration]({{ site.baseurl }}/2.0/configuration-reference/#commands-requires-version-21), making them reusable across your config.
 
 ```yaml
 #...
```
Comment (thomasr8) [inline jekyll/_cci2/concepts.md:170]: Do we need a version number here?

File: jekyll/_cci2/concepts.md [hunk 3]
```diff
@@ -158,9 +193,66 @@ Workflows define a list of jobs and their run order. It is possible to run jobs
...
+          requires:
+           - build1 # wait for build1 job to complete successfully before starting
+           # run build2 and build3 concurrently to save time.
+```
+{% endraw %}
+
+{:.tab.workflows-example.Server}
 {% raw %}
 ```yaml
 version: 2
```
Comment (thomasr8) [inline jekyll/_cci2/concepts.md:254]: Should this be 2.1?

File: jekyll/_cci2/concepts.md [hunk 4]
```diff
@@ -217,10 +309,43 @@ workflows:
...
+    steps:
+      - restore_cache: # Restores the cached dependency.
+          key: v1-repo-{{ .Environment.CIRCLE_SHA1 }}       
+```
+{% endraw %}
+
+{:.tab.cache.Server}
+{% raw %}
 ```yaml
 version: 2
```
Comment (thomasr8) [inline jekyll/_cci2/concepts.md:347]: version 2.1?

File: jekyll/_cci2/concepts.md [hunk 5]
```diff
@@ -246,16 +371,51 @@ jobs:
...
+{% raw %}
+```yaml
+version: 2.1
+
+jobs:
+  build1:
+#...   
+    steps:    
+      - persist_to_workspace: # Persist the specified paths (workspace/echo-output)
+      # into the workspace  for use in downstream job. Must be an absolute path,
```
Comment (thomasr8) [inline jekyll/_cci2/concepts.md:390]:
```suggestion
      # into the workspace for use in downstream job. Must be an absolute path,
```

File: jekyll/_cci2/concepts.md [hunk 6]
```diff
@@ -246,16 +371,51 @@ jobs:
...
+          path: /tmp/artifact-1
+          destination: artifact-file
+#...
+```        
+{% endraw %}
+
+{:.tab.workspace.Server}
 {% raw %}
 ```yaml
 version: 2
```
Comment (thomasr8) [inline jekyll/_cci2/concepts.md:417]: version 2.1?

File: jekyll/_cci2/jobs-steps.md [hunk 7]
```diff
@@ -18,58 +18,69 @@ Orbs are packages of config that you either import by name or configure inline t
...
 ```
 
 ## Sample Configuration with Concurrent Jobs
 
 Following is a sample 2.0 `.circleci/config.yml` file.
 
 {% raw %}
-```
+```yaml
 version: 2
```
Comment (thomasr8) [inline jekyll/_cci2/jobs-steps.md:72]: version 2.1?

File: jekyll/_cci2/concepts.md (hunk 1 above)
Comment (rosieyohannan) [reply jekyll/_cci2/concepts.md:131]: Ah yes this is tabbed content, and this is the server tab, so needs to stay at 2.0

File: jekyll/_cci2/concepts.md (hunk 3 above)
Comment (rosieyohannan) [reply jekyll/_cci2/concepts.md:254]: same as above

File: jekyll/_cci2/concepts.md (hunk 4 above)
Comment (rosieyohannan) [reply jekyll/_cci2/concepts.md:347]: ^

File: jekyll/_cci2/concepts.md (hunk 5 above)
Comment (rosieyohannan) [reply jekyll/_cci2/concepts.md:390]: 👍

File: jekyll/_cci2/concepts.md (hunk 6 above)
Comment (rosieyohannan) [reply jekyll/_cci2/concepts.md:417]: ^

File: jekyll/_cci2/concepts.md (hunk 2 above)
Comment (rosieyohannan) [reply jekyll/_cci2/concepts.md:170]: As it's an extract of the config we don't need the version

**Pull request merged**